go-runtime/encoding/encoding.go

```diff
@@ -31,37 +38,34 @@ func encodeValue(v reflect.Value, w *bytes.Buffer) error {
 		w.WriteString("null")
 		return nil
 	}
+
 	t := v.Type()
-	switch {
-	case t.Kind() == reflect.Ptr && t.Elem().Implements(jsonMarshaler):
-		v = v.Elem()
-		fallthrough
 
-	case t.Implements(jsonMarshaler):
-		enc := v.Interface().(json.Marshaler) //nolint:forcetypeassert
-		data, err := enc.MarshalJSON()
+	// Special-cased types
+	switch {
+	case t == reflect.TypeFor[time.Time]():
+		data, err := json.Marshal(v.Interface().(time.Time))
```

Comment:
We can remove the cast here and below. I originally fixed it [here](https://github.com/TBD54566975/ftl/pull/1424) after Alec's comment but reverted that in order to revert the big one

Reply:
oop thanks, added back!